### DISCUSSION TITLE
Can you calculate the total "real particles" energy of a species just from vx output?

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#748] [Q&A] [answered]
Hello there,

This question might be related to https://github.com/Warwick-Plasma/epoch/discussions/712 

I have a beam species with KE primarily in the x-direction (1D), and I'm currently outputting vx. I'm trying to calculate the **total energy of this species**, and I'm wondering if I need to output the particle weights to do this accurately, or if I can determine the total energy just from vx alone (and scale it by some weighting factor?).

Thanks again!

**Comment by Status-Mirror** [comment, marked as answer]
Hi @wnguyen1312,

To do this properly, you would need `vx`, `vy`, `vz` and `weight` outputs. Even if your particles mostly move in the $x$ direction, you shouldn't ignore $y$ and $z$ contributions if you want an accurate estimate for total energy. If your macro-particles have different weights, summing the energy contributions of the macro-particles would give you the wrong answer, and no weighting-factor scaling could correct for this.

Cheers,
Stuart
